Document/0x06f-Testing-Local-Authentication.md

```diff
@@ -9,26 +9,196 @@ As a result only requirement "4.6	Biometric authentication, if any, is not event
+`LAPolicy` has two options:
+	* `deviceOwnerAuthentication`(Swift) or `LAPolicyDeviceOwnerAuthentication`(Objective-C) - using Touch ID or the device password
+	* `deviceOwnerAuthenticationWithBiometrics` (Swift) or `LAPolicyDeviceOwnerAuthenticationWithBiometrics`(Objective-C) - use Touch ID only
+
+Appropriate error will be returned to completion handler in following cases: authentication failed, user cancelled authentication, user have chosen fallback, system cancelled authentication, passcode is not set on device, TouchID is not available, TouchID is not enrolled.
+
+###### Swift
+```
+let context = LAContext()
+var error: NSError?
+
+guard context.canEvaluatePolicy(.deviceOwnerAuthentication, error: &error) else {
+	// Could not evaluate policy; look at error and present an appropriate message to user
+}
+
+context.evaluatePolicy(.deviceOwnerAuthentication, localizedReason: "Please, pass authorisation to enter this area") { success, evaluationError in
+	guard success else {
+		// User did not authenticate successfully, look at evaluationError and take appropriate action
+	}
+
+	// User authenticated successfully, take appropriate action
+}
+```
+
+###### Objective-C
+```
+LAContext *myContext = [[LAContext alloc] init];
+NSError *authError = nil;
+
+if ([myContext canEvaluatePolicy:LAPolicyDeviceOwnerAuthenticationWithBiometrics error:&authError]) {
+    [myContext evaluatePolicy:LAPolicyDeviceOwnerAuthenticationWithBiometrics
+                  localizedReason:@"Please, pass authorisation to enter this area"
+                            reply:^(BOOL success, NSError *error) {
+            if (success) {
+                // User authenticated successfully, take appropriate action
+            } else {
+                // User did not authenticate successfully, look at error and take appropriate action
+            }
+        }];
+} else {
+    // Could not evaluate policy; look at authError and present an appropriate message to user
+}
+```
+
+##### Security.framework
+Keychain allows to save items with special SecAccessControl attribute, which will allow to get item from Keychain only after user will pass Touch ID authentication (or passcode, if such fallback is allowed by attribute parameters).
+
+
+
+In following example we will save "S00p3r_haCk3r_strong_password" string to Keychain which can be accessed only on current device while passcode is set (`kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly` parameter) after Touch ID authentication for currently enrolled fingers only (`.touchIDCurrentSet parameter`):
+###### Swift
+```
+// 1. create AccessControl object that will represent authentication settings
+
+var error: Unmanaged<CFError>?
+
+guard let accessControl = SecAccessControlCreateWithFlags(kCFAllocatorDefault,
+	kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly,
+	.touchIDCurrentSet,
```

Comment:
As explained in key management, it would be nice to refer to the key-management here, as there are multiple control flags you can use here. touchIDCurrentSet is the strongest, but it might be necessary to have a lesser strong setting as well, please advice on that too in this situation.